src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp

```diff
@@ -1148,26 +1353,27 @@ namespace OpenMS
 
     //LOG_INFO << "searchMass: neutral_query_mass=" << neutral_query_mass << " diff_mz=" << diff_mz << " ppm allowed:" << mass_error_value_ << std::endl;
 
-
     // binary search for formulas which are within diff_mz distance
-    if (mass_mappings_.size() < 1)
+    if (mass_mappings_.empty())
     {
       throw Exception::InvalidValue(__FILE__, __LINE__, __PRETTY_FUNCTION__, "There are no entries found in mass-to-ids mapping file! Aborting... ", "0");
     }
 
-    std::vector<MappingEntry_>::iterator lower_it = std::lower_bound(mass_mappings_.begin(), mass_mappings_.end(), neutral_query_mass - diff_mz, CompareEntryAndMass_()); // first element equal or larger
-    std::vector<MappingEntry_>::iterator upper_it = std::upper_bound(mass_mappings_.begin(), mass_mappings_.end(), neutral_query_mass + diff_mz, CompareEntryAndMass_()); // first element greater than
+    std::vector<MappingEntry_>::const_iterator lower_it = std::lower_bound(mass_mappings_.begin(), mass_mappings_.end(), neutral_query_mass - diff_mz, CompareEntryAndMass_()); // first element equal or larger
+    std::vector<MappingEntry_>::const_iterator upper_it = std::upper_bound(mass_mappings_.begin(), mass_mappings_.end(), neutral_query_mass + diff_mz, CompareEntryAndMass_()); // first element greater than
```

Comment:
with upper_bound it would be the last non-hit of a sequence of potentially equal values bigger than neutral_query_mass + diff_mz.
using lower_bound(..., neutral_query_mass + diff_mz + epsilon) gives the first non-hit that can be used like an .end() iterator.